### ISSUE TITLE
Git Sync

### ISSUE THREAD

**Issue by cgatian** [REST-API-Client/API-Client-VSCode-Extension#66]
Ability to save request information to the Git repo so users can share configuration.

**Comment by unwls** [comment]
Interesting...

I don't quite get the whole idea since the extension already saves user request information in their local machine.

Want to enlighten me?

**Comment by cgatian** [comment]
I should try out the plugin :)
Basically, you can save your requests in a git repository so they can be shared with a team that pulls your repo.

**Comment by akash1520** [comment]
Sounds interesting, this would give this extension an edge over thunderclient, as thunder client doesn't offer this and I have suffered due to lack of a feature like this in a hackathon.

**Comment by unwls** [comment]
If anybody's willing to implement this feature, feel free to do so! It does sound like a cool feature to add

**Comment by akash1520** [comment]
I would love to give it a try, I'll keep you updated! Assign it to me if sounds fair.

**Comment by lokize** [comment]
must have feature +1

**Comment by PaulAtST** [comment]
I have been wishing there was a VS Code REST client that could easily sync to git for years. This would be a very useful feature

**Comment by cgatian** [comment]
Just found out about Bruno, going to give it a try. They also have a VSCode extension.
https://www.usebruno.com/
https://marketplace.visualstudio.com/items?itemName=bruno-api-client.bruno

**Comment by akash1520** [comment]
@unwls @Codedwells I'll try implementing this, if it's still not done

**Comment by akash1520** [comment]
### Proposal

1. Will make user get their developer access token for personal use from Github
2. Will use this token to create a separate repo for saving data of this extension
3. Their will be different directories in this repo for each project in which our extension is being used
4.  It will auto save data when user makes a requests using this extension and will also auto save data whenever users exits or closes the extension

In short, user won't ever need to press a save button or anything like that, everything will be done automatically.

Let me know if you think anything should be done differently!

**Comment by cgatian** [comment]
No, just save the files local to the repo. The plugin needs to be able to read and update these files

**Comment by unwls** [comment]
@akash1520 PR's are always welcome 👍🏻
